server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -377,14 +377,8 @@ class InteractiveSession(
 
   private val app = mockApp.orElse {
     if (livyConf.isRunningOnYarn()) {
-      val driverProcess = client.map { client =>
-        if (client.getDriverProcess == null) {
-          None
-        }
-        else {
-          Some(new LineBufferedProcess(client.getDriverProcess))
-        }
-      }.getOrElse(None)
+      val driverProcess = Option(client.map(_.getDriverProcess).get)
```

Comment:
Is it
`client.flatMap(c => Option(c.getDriverProcess)).map(new LineBufferedProcess(_))` ?